### DISCUSSION TITLE
Error 451 / Right-Holders and Prohibited Access

### DISCUSSION THREAD

**Discussion by lincoln017** [tidal-music/discussions#14] [Q&A]
When testing the API to retrieve basic metadata about an artist's albums, I'm getting many 451 status messages with the text as follows: "Unavailable due to demand from the right-holders to prohibit access to the resource".  Given that the data I'm looking for is quite generic -- title, releaseDate, number of tracks, etc. -- I'm confused why this type of high level information is protected.  Without that information, I can't enable my app to integrate successfully with the Tidal ecosystem.  

Am I missing something here?  Thanks.

**Comment by stefanmoro** [comment]
Hey @lincoln017, and welcome to the forum.

TIDAL Developer Platform is in very early stages, and our hope is that we will be able to supply more functionality and data going forward. Throughout this journey, it is important for us to work closely together with labels, artists and content owners to make sure the solutions we build are beneficial for all involved parties.

As of today we can unfortunately not expose our full catalogue via our API. This is something we are actively working on, and we hope to be able to expand the part of our catalogue we expose in the future.

Thanks

**Reply by BitesizedLion** [reply]
> As of today we can unfortunately not expose our full catalogue via our API

But why? Rights holders aren't okay with exposing the most basic metadata via Tidal's API, but are okay with it on other streaming services' apis (e.g. Spotify)?

**Reply by stefanmoro** [reply]
We  can’t speak to other services’ APIs, but are focused on improving and growing TIDAL’s API offering.

**Reply by jozefKruszynski** [reply]
@BitesizedLion My assumption is that at the end of the day it's simply the fact that agreements need to be signed in order for any information to be exposed via a new API. Just a matter of time I suppose

**Comment by aconstas** [comment]
Hi @stefanmoro,
Is there a timeline for a given percentage of catalog coverage for the API? Will the full catalog ever be available?
I want to use the API to display albums and their metadata but it is difficult when there are so many "Unavailable due to demand from the right-holders to prohibit access to the resource" responses.

**Reply by stefanmoro** [reply]
Hi @aconstas 

We are actively working on this topic, but can unfortunately not give any promises as of yet. Will post here when we have any updates.
